code-challenges.test.js

```diff
@@ -2,58 +2,232 @@
+//
+// I used map() => to go through each word in words array and..
+// ensure each word is lowercase then stored to be used later..
+// then we want to apply the toUpperCase() to just the first letter in each word
+// I use the charAt() method to return the char at index 0 to accomplish this
+// then return the combined capital letter and lowercase letters using
+// slice(1) and concatenation
+//
+// **Note** since using map() this will not mutate the original array.
+//------------------------------------------------------------------------------
+const capitalize = (words) => {
+  return words.map((word) => {
+    const lowercase = word.toLowerCase();
+    return word.charAt(0).toUpperCase() + lowercase.slice(1);
+  });
+};
+
+// =============================================================================
+// --------------------3) Create a function that takes in a string and logs the
+// index of the first vowel.
+// =============================================================================
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
+// =============================================================================
+
+//---------------------== Function Explanation ==-------------------------------
+// I decided to use two it-statements to be more descriptive on this test since
+// I wanted to test for an edge case for when a word may not have a vowel.
+//
+// ---- Initial Test: ReferenceError: getFirstVowel is not defined -------------
+//------------------------------------------------------------------------------
+
+describe("getFirstVowel", () => {
+  it("returns the index of the first vowel", () => {
+    expect(getFirstVowel(vowelTester1)).toEqual(1);
+    expect(getFirstVowel(vowelTester2)).toEqual(0);
+    expect(getFirstVowel(vowelTester3)).toEqual(2);
+  });
+  it("returns null if the word does not contain a vowel", () => {
+    expect(getFirstVowel(vowelTester4)).toEqual(null);
+  });
+  it("returns the index of the first vowel if it is capital", () => {
+    expect(getFirstVowel(vowelTester5)).toEqual(1);
+  });
+});
+
+// -----------------=== Provided Test Cases ===---------------------------------
+const vowelTester1 = "learn";
 // Expected output: 1
-const vowelTester2 = "academy"
+const vowelTester2 = "academy";
 // Expected output: 0
-const vowelTester3 = "challenges"
+const vowelTester3 = "challenges";
 // Expected output: 2
 
+// -----------------== Additional Edge Cases ==---------------------------------
+const vowelTester4 = "myth";
+// Expected output: null
+const vowelTester5 = "LEARN";
+
+// Expected output: null
+// =============================================================================
 // b) Create the function that makes the test pass.
+// =============================================================================
+
+//---------------------== Function Explanation ==-------------------------------
+// Input: String
```

Comment:
Excellent work in this section!  Your pseudocode is excellent and very easy to read, and your tests, code and logic look great!  If you want to stretch yourself, consider using a higher-order function for this last problem.